pkg/roll/run_update.go

```diff
@@ -167,13 +167,44 @@ ROLL_LOOP:
 				break ROLL_LOOP
 			}
 
-			next := algorithm(oldNodes.Healthy, newNodes.Healthy, u.DesiredReplicas, u.MinimumReplicas)
+			next := rollAlgorithm(oldNodes.Healthy, newNodes.Healthy, u.DesiredReplicas, u.MinimumReplicas)
 			if next > 0 {
+				// apply the delay only if we've already added to the new RC, since there's
+				// no value in sitting around doing nothing before anything has happened.
+				if newNodes.Desired > 0 && u.RollDelay > time.Duration(0) {
+					u.logger.WithField("delay", u.RollDelay).Infof("Waiting %v before continuing deploy", u.RollDelay)
```

Comment:
I'm conflicted about duplicating the `u.RollDelay` value in these log messages. I can understand that we want the user to see "Waiting 30s ...". Why bother with the field?